pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/util/DesignerUtil.java

```diff
@@ -61,35 +123,43 @@ public static LanguageVersion getLanguageVersionFromExtension(String filename) {
             String[] tokens = filename.split("\\.");
             return extensionsToLanguage.get(tokens[tokens.length - 1]);
         }
-
         return null;
     }
 
 
-    /**
-     * Name of the designer's settings file.
-     *
-     * @return The name
-     */
-    public static File getSettingsFile() {
-        return DESIGNER_SETTINGS_FILE;
-    }
-
-
-    public static LanguageVersion[] getSupportedLanguageVersions() {
+    public static List<LanguageVersion> getSupportedLanguageVersions() {
         if (supportedLanguageVersions == null) {
             List<LanguageVersion> languageVersions = new ArrayList<>();
             for (LanguageVersion languageVersion : LanguageRegistry.findAllVersions()) {
-                LanguageVersionHandler languageVersionHandler = languageVersion.getLanguageVersionHandler();
-                if (languageVersionHandler != null) {
-                    Parser parser = languageVersionHandler.getParser(languageVersionHandler.getDefaultParserOptions());
-                    if (parser != null && parser.canParse()) {
-                        languageVersions.add(languageVersion);
-                    }
-                }
+                Optional.ofNullable(languageVersion.getLanguageVersionHandler())
+                        .map(handler -> handler.getParser(handler.getDefaultParserOptions()))
+                        .filter(Parser::canParse)
+                        .ifPresent(p -> languageVersions.add(languageVersion));
             }
-            supportedLanguageVersions = languageVersions.toArray(new LanguageVersion[languageVersions.size()]);
+            supportedLanguageVersions = languageVersions;
         }
         return supportedLanguageVersions;
     }
+
+
+    /**
+     * Binds the underlying property to a source of values. The source property is also initialised using the setter.
+     *
+     * @param underlying The underlying property
+     * @param ui         The property exposed to the user (the one in this wizard)
+     * @param setter     Setter to initialise the UI value
+     * @param <T>        Type of values
+     */
+    public static <T> void rewire(Property<T> underlying, ObservableValue<? extends T> ui, Consumer<? super T> setter) {
```

Comment:
Are the static methods of [Bindings](https://docs.oracle.com/javase/8/javafx/api/javafx/beans/binding/Bindings.html) of some help here?

Reply:
Not really, Bindings is about creating new ObservableValues by composing existing ones